### PR TITLE
feat: finalise

### PR DESCRIPTION
**Pull request by nchamo**
We are now implementing `finalise`. This function will:
* Check if the position can be terminated (it's finished or swaps are paused)
* Terminate the position
* Unwrap unswapped funds (if needed)
* Unwrap swapped funds (if needed)

_Note: I've only added unit tests that verify revert flows. Will add a good E2E test in another PR_

### CODE REVIEW

File: src/interfaces/mean/IDCAHub.sol [hunk 1]
```diff
@@ -39,13 +19,6 @@ interface IDCAHub {
     Permission[] permissions;
   }
   
-  /**
-   * @notice Returns a user position
-   * @param positionId The id of the position
-   * @return position The position itself
-   */
-  function userPosition(uint256 positionId) external view returns (UserPosition memory position);
```
Comment (nchamo) [inline src/interfaces/mean/IDCAHub.sol:47]: Realized we didn't need this after all

File: src/bridges/mean/MeanBridge.sol [hunk 2]
```diff
@@ -95,6 +96,53 @@ contract MeanBridge is BridgeBase, Ownable2Step {
...
+
+        // Terminate position and clean things up
+        uint256 _positionId = positionByNonce[_interactionNonce];
+        delete positionByNonce[_interactionNonce];
+        (uint256 _unswapped, uint256 _swapped) = DCA_HUB.terminate(_positionId, THIS_ADDRESS, THIS_ADDRESS);
+
+        if (_unswapped > 0) {
+            if (!DCA_HUB.paused()) {
+                // If there are still unswapped funds, then we will only allow closing the DCA position
+                // if swaps have been paused
```
Comment (0xged) [inline src/bridges/mean/MeanBridge.sol]: This is a limitation for shared positions, right? If we had 1 position p/user we ould allow this, right?

File: src/bridges/mean/MeanBridge.sol [hunk 3]
```diff
@@ -95,6 +96,53 @@ contract MeanBridge is BridgeBase, Ownable2Step {
         return (0, 0, true);
     }
 
+    /**
+     * @notice A function which will allow the users to close their DCA positions on Mean Finance
```
Comment (0xged) [inline src/bridges/mean/MeanBridge.sol:100]: Maybe we should add for `@dev` that position users will only be able to close position IF protocol is paused or position has ended.

File: src/bridges/mean/MeanBridge.sol (hunk 2 above)
Comment (nchamo) [reply src/bridges/mean/MeanBridge.sol]: We can't tell when that's the case, so users will only be able to terminate their position when it's finished

**Pull request merged**